java/src/main/java/com/azure/graph/bulk/impl/model/GremlinEdgeVertexInfo.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.graph.bulk.impl.model;
+
+import com.azure.graph.bulk.impl.annotations.GremlinId;
+import com.azure.graph.bulk.impl.annotations.GremlinPartitionKey;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.reflect.FieldUtils;
+
+import java.lang.reflect.Field;
+import java.util.Objects;
+
+@Data
+@Builder
+@AllArgsConstructor
+public class GremlinEdgeVertexInfo {
+    private String id;
+    private String label;
+    private String partitionKey;
```

Comment:
The partition key can be non-string. It can be a number or even a boolean. Basically any Cosmos DB primitive data type.

Reply:
Consider making this field `GremlinPartitionKey` and then make `GremlinPartitionKey` support non-string partition values